enterprise/kernel/src/main/java/org/neo4j/kernel/enterprise/builtinprocs/EnterpriseBuiltInDbmsProcedures.java

```diff
@@ -245,6 +245,24 @@ public ProcedureResult( ProcedureSignature signature )
         }
     }
 
+    @Description( "List the active lock requests granted for the transaction executing the query with the given query id." )
+    @Procedure( name = "dbms.listActiveLocks", mode = DBMS )
+    public Stream<ActiveLocksQueryResult> listActiveLocks( @Name( "queryId" ) String queryId )
+            throws InvalidArgumentsException
+    {
+        try
+        {
+            long id = fromExternalString( queryId ).kernelQueryId();
+            return getActiveTransactions( tx -> executingQueriesWithId( id, tx ) )
+                    .flatMap( pair -> pair.first().activeLocks().stream().map( ActiveLocksQueryResult::new ) );
+        }
+        catch ( UncaughtCheckedException uncaught )
+        {
+            throwIfPresent( uncaught.getCauseIfOfType( InvalidArgumentsException.class ) );
+            throw uncaught;
+        }
```

Comment:
I implemented it as snapshots since that was easier than streaming on the implementation level. Initially I had streams on the API level but then asked myself "who am I fooling", and changed the API to collections like it was in the implementation.

We need to convert to something with the appropriate public fields, not sure we want to have the internal API use such style of object, but maybe we do. Or perhaps we can turn this into some sort of visitor based (internal) API then have that directly construct the stream of the appropriate objects. That would be a more complicated API, but would imply less allocation.